File: tests/grab.cpp
#include <stdio.h>
#include <opencv2/opencv.hpp>
 
using namespace cv;
using namespace std;
 
int main(int argc,char ** argv)
{
  VideoCapture cap(0);
  if (!cap.isOpened()) {
    cerr << "ERROR: Unable to open the camera" << endl;
    return 0;
  }
 
  Mat frame;
  cout << "Start grabbing, press a key on Live window to terminate" << endl;
  while(1) {
    cap >> frame;
    if (frame.empty()) {
        cerr << "ERROR: Unable to grab from the camera" << endl;
        break;
    }
    imshow("Live",frame);
    if (waitKey(1)>=0)
      break;
  }
 
  cout << "Closing the camera" << endl;
  cap.release();
  destroyAllWindows();
  cout << "Done." <<endl;
  return 0;
}

File: tests/grab.py
import numpy as np
import cv2
import os
from imutils.video import FPS
os.system("sudo modprobe bcm2835-v4l2")
fps = FPS().start()

cap = cv2.VideoCapture(0)
if cap.isOpened() == False:
    print('ERROR: Unable to open the camera')
else:
    print('Start grabbing, press a key on Live window to terminate')
    cv2.namedWindow('Live');
    while( cap.isOpened() ):
        ret,frame = cap.read()
        if ret==False:
            print('ERROR: Unable to grab from the camera')
            break;

        cv2.imshow('Live ({:.2f})'.format(fps.fps()),frame)
        fps.update()
        if cv2.waitKey(1) >= 0:
           break
    print('Closing the camera')

cap.release()
cv2.destroyAllWindows()
# stop the timer and display FPS information
fps.stop()
print("[INFO] elasped time: {:.2f}".format(fps.elapsed()))
print("[INFO] approx. FPS: {:.2f}".format(fps.fps()))
print('Done\n')
